cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/AnalyzerUtils.java

```diff
@@ -226,13 +226,17 @@ public static int compare(double d1, double d2, double epsilon) {
   /**
    * Get a goal map with goal name as the keys.
    */
-  public static Map<String, Goal> getGoalsMapByName(KafkaCruiseControlConfig config) {
+  public static Map<String, Goal> getCaseInsensitiveGoalsByName(KafkaCruiseControlConfig config)
+      throws AnalysisInputException {
```

Comment:
This should probably be a ConfigurationException thrown at the startup time so users can know the configured goals have a problem immediately.